src/builder/edit_current_application_info.rs

```diff
@@ -0,0 +1,177 @@
+    #[serde(skip_serializing_if = "Option::is_none")]
+    flags: Option<ApplicationFlags>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    icon: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cover_image: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    interactions_endpoint_url: Option<Cow<'a, str>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    tags: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_webhooks_url: Option<Cow<'a, str>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_webhook_status: Option<EventWebhookStatus>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_webhook_types: Option<Vec<EventWebhookType>>,
+}
+
+impl<'a> EditCurrentApplicationInfo<'a> {
+    /// Creates a new builder instance with all values set to None.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Default custom authorization URL for the app, if enabled.
+    pub fn install_url(mut self, url: impl Into<Cow<'a, str>>) -> Self {
+        self.custom_install_url = Some(url.into());
+        self
+    }
+
+    /// Description of the app.
+    pub fn description(mut self, description: impl Into<Cow<'a, str>>) -> Self {
+        self.description = Some(description.into());
+        self
+    }
+
+    /// Role connection verification URL for the app.
+    pub fn verification_url(mut self, url: impl Into<Cow<'a, str>>) -> Self {
+        self.role_connections_verification_url = Some(url.into());
+        self
+    }
+
+    /// Settings for the app's default in-app authorization link, if enabled.
+    pub fn install_params(mut self, params: CreateInstallParams) -> Self {
+        self.install_params = Some(params);
+        self
+    }
+
+    /// Default scopes and permissions for each supported installation context.
+    /// Value for each key is an integration type configuration object.
+    pub fn integration_types_config(
+        mut self,
+        config: HashMap<InstallationContext, InstallationContextConfig>,
+    ) -> Self {
+        self.integration_types_config = Some(config);
+        self
+    }
+
+    /// App's public flags.
+    ///
+    /// Only limited intent flags (GATEWAY_PRESENCE_LIMITED, GATEWAY_GUILD_MEMBERS_LIMITED,
+    /// and GATEWAY_MESSAGE_CONTENT_LIMITED) can be updated via the API.
+    pub fn flags(mut self, flags: ApplicationFlags) -> Self {
+        self.flags = Some(flags);
+        self
+    }
+
+    /// Icon for the app
+    pub fn icon(mut self, base64_image_data: String) -> Self {
+        self.icon = Some(base64_image_data);
+        self
+    }
+
+    /// Sets the default rich presence invite cover image using base64 image data.
+    pub fn cover_image(mut self, base64_image_data: String) -> Self {
+        self.cover_image = Some(base64_image_data);
+        self
+    }
```

Comment:
Rather than passing in base64 data manually as a `String`, we should allow the user to decide the place they want to source the image from and convert it to base64 on their behalf, similar to how [`EditProfile::avatar`](https://github.com/serenity-rs/serenity/blob/current/src/builder/edit_profile.rs#L49-L52) does it using `CreateAttachment`.